webui/ocw/lib/azure.py

```diff
@@ -71,3 +79,79 @@ def list_resource_groups(self):
 
     def delete_resource(self, resource_id):
         return self.resource_mgmt_client().resource_groups.delete(resource_id)
+
+    def cleanup_all(self):
+        ''' Cleanup all autodateed data which might created during automated tests.'''
+        resourcegroup = self.cfgGet('cleanup', 'azure-storage-resourcegroup')
+        storage_account = self.cfgGet('cleanup', 'azure-storage-account-name')
+        storage_client = StorageManagementClient(self.sp_credentials(), self.subscription())
+        storage_keys = storage_client.storage_accounts.list_keys(resourcegroup, storage_account)
+        storage_keys = [v.value for v in storage_keys.keys]
+        block_blob_service = BlockBlobService(account_name='openqa', account_key=storage_keys[0])
+
+        containers = block_blob_service.list_containers()
+        for c in containers:
+            self.__logger.debug('Found container {}'.format(c.name))
+            if (re.match('^bootdiagnostics-', c.name)):
+                self.cleanup_bootdiagnostics_container(block_blob_service, c)
+            if (c.name == 'sle-images'):
+                self.cleanup_sle_images_container(block_blob_service, c)
+
+    def cleanup_bootdiagnostics_container(self, bbsrv, container):
+        timeout = datetime.now(timezone.utc) + timedelta(seconds=60*60*24)
+        last_modified = container.properties.last_modified
+        generator = bbsrv.list_blobs(container.name)
```

Comment:
can you explain this variable name ?

Reply:
https://azure-storage.readthedocs.io/ref/azure.storage.blob.baseblobservice.html#azure.storage.blob.baseblobservice.BaseBlobService.list_containers

```
Returns a generator to list the containers...
````